### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by lmontrieux**
# Upgrade most dependencies to latest version

> Zalando ticket : ARUHA-1683

## Description
This PR upgrades (almost) all dependencies, according to the following plan:
- upgrade Spring boot to latest stable patch version
- upgrade all spring boot dependencies to:
  - the same version as used in spring boot, if the previous version was lower
  - unchanged, if the version used was already more recent than the spring boot dependency
- upgrade other dependencies to the latest stable version whenever possible
- upgrade test dependencies to the latest stable version that does not break tests

Exception: Zalando Problem libraries will be upgraded in a separate PR, as they require significant refactoring.

## Review
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

### CODE REVIEW

**Comment by adyach** [comment]
@lmontrieux it will be nice to have gradle upgraded as well :)

**Comment by lmontrieux** [comment]
@adyach upgraded gradle to 4.7 in 1a34647

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+plugins {
+    id 'java'
+    id 'groovy'
+    id 'eclipse'
+    id 'application'
+    id 'jacoco'
+    id 'findbugs'
+    id 'checkstyle'
+    id 'project-report'
+    id 'org.springframework.boot' version "1.5.13.RELEASE"
```
Comment (adyach) [inline build.gradle]: does "$springBootVersion" work here ?

**Comment by adyach** [comment]
👍

**Comment by lmontrieux** [comment]
👍

**Comment by lmontrieux** [comment]
Deploy validation please

File: build.gradle (hunk 1 above)
Comment (lmontrieux) [reply build.gradle]: I tried, but apparently, no

**Comment by lmontrieux** [comment]
Deploy validation please

**Comment by lmontrieux** [comment]
Validation deployment fails with:
```
[WARN] [null] [main] [org.springframework.boot.context.embedded.AnnotationConfigEmbeddedWebApplicationContext] --- Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'eventTypeDbRepository' defined in URL [jar:file:/nakadi.jar!/BOOT-INF/classes!/org/zalando/nakadi/repository/db/EventTypeDbRepository.class]: Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jacksonObjectMapper' defined in class path resource [org/zalando/nakadi/config/JsonConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.fasterxml.jackson.databind.ObjectMapper]: Factory method 'jacksonObjectMapper' threw exception; nested exception is java.lang.VerifyError: Cannot inherit from final class
```

File: build.gradle (hunk 1 above)
Comment (antban) [reply build.gradle]: No, it is not working in plugins

**Comment by antban** [comment]
@lmontrieux I see that it is validating now, but it have to inclure all the changes from master

**Comment by antban** [comment]
:+1:

**Comment by v-stepanov** [comment]
comment

**Comment by lmontrieux** [comment]
First comment

**Comment by antban** [comment]
:+1:

**Comment by lmontrieux** [comment]
👍

**Comment by lmontrieux** [comment]
Deploy validation please.

**Comment by antban** [comment]
deploy validation please

**Comment by lmontrieux** [comment]
Deploy validation please

**Comment by lmontrieux** [comment]
deploy validation please

**Comment by lmontrieux** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by antban** [comment]
:+1:

**Comment by antban** [comment]
:+1:

**Comment by adyach** [comment]
👍

**Comment by antban** [comment]
:+1:

**Pull request merged**